vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/resources/META-INF/resources/frontend/datepickerConnector.js

```diff
@@ -152,9 +158,14 @@ import { extractDateParts, parseDate as _parseDate } from '@vaadin/date-picker/s
           const formatterAndParser = createFormatterAndParser(usedFormats);
 
           // Merge current web component I18N settings with new I18N settings and the formatting and parsing functions
-          // Defer setting I18N property into microtask as a workaround for: https://github.com/vaadin/flow-components/issues/4500
-          queueMicrotask(() => {
-            datepicker.i18n = Object.assign({}, datepicker.i18n, i18n, formatterAndParser);
+          const updatedI18n = Object.assign({}, datepicker.i18n, i18n, formatterAndParser);
+          datepicker.i18n = updatedI18n;
+          // Update I18N property again after the element is ready as a workaround for both:
+          // https://github.com/vaadin/flow-components/issues/4500
+          // https://github.com/vaadin/flow-components/issues/4667
+          // This workaround is only necessary for v23, and will be removed in v24.
+          datepicker.addEventListener("date-picker-ready", () => {
```

Comment:
What will happen if you call `DatePicker.setLocale(...)` after the ready callback has been already invoked? I'm afraid the i18n update will never happen in that case.

Reply:
I think the first update call still handles those cases. The workaround is just the extra call that updates the i18n again after the initial erroneous override.

Reply:
Aha, I see. Then my only concern would be that it adds more and more listeners that are never invoked. I don't know how frequently updatei18N is usually called – it probably depends on the application. I think it would make sense to check whether the component is ready before adding a listener.

Reply:
This was a very valid concern. However, I recently updated the PR based on Tomi's [suggestion](https://github.com/vaadin/flow-components/pull/4706#discussion_r1115859796), and now there are no extra listeners involved.